src/Text/Layout/Table.hs

```diff
@@ -245,12 +244,12 @@ checkeredCells f g = zipWith altLines $ cycle [[f, g], [g, f]]
 
 -- | Create a 'RowGroup' by aligning the columns vertically. The position is
 -- specified for each column.
-colsG :: Monoid a => [Position V] -> [Col a] -> RowGroup a
+colsG :: Cell a => [Position V] -> [Col a] -> RowGroup a
 colsG ps = rowsG . colsAsRows ps
 
 -- | Create a 'RowGroup' by aligning the columns vertically. Each column uses
 -- the same vertical positioning.
-colsAllG :: Monoid a => Position V -> [Col a] -> RowGroup a
+colsAllG :: Cell a => Position V -> [Col a] -> RowGroup a
 colsAllG p = rowsG . colsAsRowsAll p
```

Comment:
Not all things you would want to put into a table are instances of `Monoid`, and this would preclude their use with this function. In particular, the proposed `ElidableList` in one of the draft PRs does not have a `Monoid` instance.

The only thing the monoid constraint is needed for padding, for which `mempty` is used. Under the assumption that one would only use this to later be rendered, we would already have a `Cell` instance and so may as well use the new `emptyCell` element of the typeclass.

That said, the most general solution may be to take the padding as an argument: `colsGWith :: a -> [Position V] -> [Col a] -> RowGroup a`. We could leave the existing `colsG = colsGWith mempty`, but I'm not sure this would be worth the effort.

Reply:
Just a follow-up that this ‘most general solution’ is actually implemented already as `vPad` and `vPadAll`. These functions are just convenience calls to that.

Reply:
I am happy with your initial explanation already.